src/pl.x64.json

```diff
@@ -365,9 +365,10 @@
 		{
 			"folder-name": "GedcomLexer",
 			"display-name": "GEDCOM Lexer",
-			"version": "0.4.0.140",
-			"id": "d1c492beccc3531efebd3c98a4e3159380283b6b84e55812979861a3d791c6ab",
-			"repository": "https://sourceforge.net/projects/gedcomlexer/files/GedcomLexer-0.4.0-r140/GedcomLexer-0.4.0-r140-x64.zip",
+			"version": "0.5.0.170",
+			"npp-compatible-versions": "[7.7,]",
```

Comment:
Is that correct minimum version for the x64 release?
It seems [r151][0] aligned the trunk with Npp 8.4.2, with at least two breaking changes:

* `Sci_PositionCR` [is now][1] of type `intptr_t`
* the `SCI_SETLEXER` message [is gone][2]

I would suggest 8.4 as the very oldest safe version.
You could say 8.4.2 to be completely safe, but Scintilla's interface didn't change much in that release, and 8.4 is a more obvious milestone.


[0]: https://sourceforge.net/p/gedcomlexer/code/151
[1]: https://sourceforge.net/p/gedcomlexer/code/151/tree//trunk/src/scintilla_include/Sci_Position.h?diff=52e08bfac4d1043fb4b929a9:150
[2]: https://sourceforge.net/p/gedcomlexer/code/151/tree//trunk/src/scintilla_include/Scintilla.h?diff=52e08bfac4d1043fb4b929a9:150

Reply:
@rdipardo Thanks for your comments. I was a little concerned about where to set the lower version threshold. I'll explain my choice and see if you think it makes sense.
First, Sci_PositionCR and SCI_SETLEXER are not used by this plugin. The plugin was revised for ILexer4 in 2019 which corresponds to version 7.7 of Npp. Other than the Npp interface changes, ILexer5 did not affect the lexer code. My testing of this release included Npp 7.8.1 so I extrapolated to 7.7.

Reply:
@rdipardo I just read [Prevent plugin from crash on v8.3 and later version: Recompile x64 plugins with new header](https://community.notepad-plus-plus.org/topic/22471/prevent-plugin-from-crash-on-v8-3-and-later-version-recompile-x64-plugins-with-new-header/1). Now I understand the need for Sci_PositionCR for files > 2GB. After the necessary changes, I'll target 8.4 for the plugin's earliest Npp version. Thanks for the head's up!